Makefile

```
@@ -11,25 +13,167 @@ HELM_BUCKET 	?= "gs://scylla-operator-charts"
 HELM_LOCAL_REPO	?= "helm/repo"
 HELM_REPOSITORY	?= "https://scylla-operator-charts.storage.googleapis.com"
 
-.EXPORT_ALL_VARIABLES:
-DOCKER_BUILDKIT		:= 1
-GOVERSION			:= $(shell go version)
-GOPATH				:= $(shell go env GOPATH)
-KUBEBUILDER_ASSETS	:= $(GOPATH)/bin
-PATH				:= $(GOPATH)/bin:$(PATH):
-
-# Default package
-PKG := ./pkg/...
 
-# Run tests
-test: fmt vet
-	go test $(PKG) -coverprofile cover.out
-
-integration-test:
-	go test $(PKG) -tags integration
+GO_REQUIRED_MIN_VERSION ?=1.15.7
+
+GIT_TAG ?=$(shell git describe --long --tags --abbrev=7 --match 'v[0-9]*' || echo 'v0.0.0-unknown')
+GIT_COMMIT ?=$(shell git rev-parse --short "HEAD^{commit}" 2>/dev/null)
+GIT_TREE_STATE ?=$(shell ( ( [ ! -d ".git/" ] || git diff --quiet ) && echo 'clean' ) || echo 'dirty')
+
+GO ?=go
+GOPATH ?=$(shell $(GO) env GOPATH)
+GOOS ?=$(shell $(GO) env GOOS)
+GOEXE ?=$(shell $(GO) env GOEXE)
+GOFMT ?=gofmt
+GOFMT_FLAGS ?=-s -l
+
+GO_VERSION :=$(shell $(GO) version | sed -E -e 's/.*go([0-9]+.[0-9]+.[0-9]+).*/\1/')
+GO_PACKAGE ?=$(shell $(GO) list -m -f '{{ .Path }}' || echo 'no_package_detected')
+GO_PACKAGES ?=./...
+
+go_packages_dirs :=$(shell $(GO) list -f '{{ .Dir }}' $(GO_PACKAGES) || echo 'no_package_dir_detected')
+GO_TEST_PACKAGES ?=$(GO_PACKAGES)
+GO_BUILD_PACKAGES ?=./pkg/cmd/...
+GO_BUILD_PACKAGES_EXPANDED ?=$(shell $(GO) list $(GO_BUILD_PACKAGES))
+go_build_binaries =$(notdir $(GO_BUILD_PACKAGES_EXPANDED))
+GO_BUILD_FLAGS ?=-trimpath
+GO_BUILD_BINDIR ?=
+GO_LD_EXTRAFLAGS ?=
+GO_TEST_PACKAGES :=./pkg/... # ./cmd/...
+GO_TEST_FLAGS ?=-race
+
+
+define version-ldflags
+-X $(1).versionFromGit="$(GIT_TAG)" \
+-X $(1).commitFromGit="$(GIT_COMMIT)" \
+-X $(1).gitTreeState="$(GIT_TREE_STATE)" \
+-X $(1).buildDate="$(shell date -u +'%Y-%m-%dT%H:%M:%SZ')"
+endef
+GO_LD_FLAGS ?=-ldflags "$(call version-ldflags,$(GO_PACKAGE)/pkg/version) $(GO_LD_EXTRAFLAGS)"
+
+# TODO: look into how to make these local to the targets
+export DOCKER_BUILDKIT :=1
+export GOVERSION :=$(shell go version)
+export KUBEBUILDER_ASSETS :=$(GOPATH)/bin
+export PATH	:=$(GOPATH)/bin:$(PATH):
```

```suggestion
export PATH :=$(GOPATH)/bin:$(PATH):
```
inconcistent whitespaces

---

weird, those were preexisting lines but my IDE showed it as one space